src/org/wikibase/data/Claim.java

```diff
@@ -0,0 +1,103 @@
+/**
+*  This program is free software; you can redistribute it and/or
+*  modify it under the terms of the GNU General Public License
+*  as published by the Free Software Foundation; either version 3
+*  of the License, or (at your option) any later version. Additionally
+*  this file is subject to the "Classpath" exception.
+*
+*  This program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with this program; if not, write to the Free Software Foundation,
+*  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+package org.wikibase.data;
+
+/**
+ * A Wikidata claim.
+ * 
+ * @author acstroe
+ *
+ */
+public class Claim {
+
+    private String id;
+    private String type;
+    private Rank rank;
+    private Property property;
+
+    private WikibaseDataType value;
+
+    public WikibaseDataType getValue() {
+        return value;
+    }
+
+    public void setValue(WikibaseDataType value) {
+        this.value = value;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((id == null) ? 0 : id.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
```

Comment:
I generated this code with Eclipse. I slightly agree with it, since `obj instanceof Claim` is not equivalent to `getClass() == obj.getClass()`. The latter works better, I think, in an inheritance context. Although it is true that for now we do not have classes that extend Claim.